### PR TITLE
[Merged by Bors] - feat: More `rpow` lemmas

### PR DESCRIPTION
**Pull request by YaelDillies**
A bunch of easy lemmas about `Real.pow` and the golf of existing lemmas with them.

Also rename `log_le_log` to `log_le_log_iff` and `log_le_log'` to `log_le_log`. Those misnames caused several proofs to bother with side conditions they didn't need.

From LeanAPAP

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Analysis/SpecialFunctions/Pow/Real.lean [hunk 1]
```diff
@@ -58,15 +59,31 @@ theorem rpow_def_of_pos {x : ℝ} (hx : 0 < x) (y : ℝ) : x ^ y = exp (log x *
...
 #align real.exp_mul Real.exp_mul
 
+@[simp, norm_cast]
+theorem rpow_int_cast (x : ℝ) (n : ℤ) : x ^ (n : ℝ) = x ^ n := by
+  simp only [rpow_def, ← Complex.ofReal_zpow, Complex.cpow_int_cast, Complex.ofReal_int_cast,
+    Complex.ofReal_re]
+#align real.rpow_int_cast Real.rpow_int_cast
+
+@[simp, norm_cast]
+theorem rpow_nat_cast (x : ℝ) (n : ℕ) : x ^ (n : ℝ) = x ^ n := by simpa using rpow_int_cast x n
```
Comment (alexjbest) [inline Mathlib/Analysis/SpecialFunctions/Pow/Real.lean:69]: Whats with these being `nat_cast` vs `natCast` elsewhere?

Comment (YaelDillies) [reply Mathlib/Analysis/SpecialFunctions/Pow/Real.lean:69]: We've hardly decided on a convention. I'm personally in favor of `natCast` (and the poll a few months back agrees with me) but most lemmas have been ported as `nat_cast`. This lemma is just moving, so I'm not changing its name. However I'm naming the new ones using `natCast`.

File: Mathlib/Analysis/SpecialFunctions/Log/Basic.lean [hunk 2]
```diff
@@ -141,23 +141,20 @@ theorem log_inv (x : ℝ) : log x⁻¹ = -log x := by
...
-#align real.log_le_log Real.log_le_log
+#align real.log_le_log Real.log_le_log_iff
+
+@[gcongr]
+lemma log_le_log (hx : 0 < x) (hxy : x ≤ y) : log x ≤ log y :=
+  (log_le_log_iff hx (hx.trans_le hxy)).2 hxy
 
 @[gcongr]
 theorem log_lt_log (hx : 0 < x) : x < y → log x < log y := by
   intro h
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Log/Basic.lean:154]:
```suggestion
theorem log_lt_log (hx : 0 < x) (h : x < y) : log x < log y := by
```

File: Mathlib/Analysis/SpecialFunctions/Stirling.lean [hunk 3]
```diff
@@ -226,10 +226,12 @@ theorem stirlingSeq_pow_four_div_stirlingSeq_pow_two_eq (n : ℕ) (hn : n ≠ 0)
     stirlingSeq n ^ 4 / stirlingSeq (2 * n) ^ 2 * (n / (2 * n + 1)) = Wallis.W n := by
   have : 4 = 2 * 2 := by rfl
   rw [stirlingSeq, this, pow_mul, stirlingSeq, Wallis.W_eq_factorial_ratio]
-  simp_rw [div_pow, mul_pow]
+  simp_rw [div_pow, mul_pow, ← exp_nsmul]
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Stirling.lean:229]:
```suggestion
  simp_rw [div_pow, mul_pow]
```

File: Mathlib/Analysis/SpecialFunctions/Stirling.lean [hunk 4]
```diff
@@ -226,10 +226,12 @@ theorem stirlingSeq_pow_four_div_stirlingSeq_pow_two_eq (n : ℕ) (hn : n ≠ 0)
...
   rw [sq_sqrt, sq_sqrt]
   any_goals positivity
-  field_simp; ring
+  simp [← exp_nsmul]
+  field_simp
```
Comment (sgouezel) [inline Mathlib/Analysis/SpecialFunctions/Stirling.lean:233]:
```suggestion
  field_simp [← exp_nsmul]
```

**Review by sgouezel — COMMENTED** [review]
bors d+
Thanks!

**Comment by YaelDillies** [comment]
bors merge

**Pull request closed without merging**